tests/integration/Project/ProjectUpdateRequestTest.php

```diff
@@ -32,293 +33,659 @@ class ProjectUpdateRequestTest extends ApiTestCase
 {
     public function testChangeName()
     {
-        $request = ProjectGetRequest::of();
-        $response = $request->executeWithClient($this->getClient());
-        $project = $request->mapResponse($response);
+        $client = $this->getApiClient();
 
-        $this->assertInstanceOf(Project::class, $project);
+        ProjectFixture::withProject(
+            $client,
+            function () use ($client) {
+                $request = RequestBuilder::of()->project()->get();
+                $response = $this->execute($client, $request);
+                $project = $request->mapFromResponse($response);
 
-        $oldName = $project->getName();
-        $name = $this->getTestRun() . '-new-name';
+                $this->assertInstanceOf(Project::class, $project);
 
-        $request = ProjectUpdateRequest::ofVersion($project->getVersion());
-        $request->addAction(ProjectChangeNameAction::ofName($name));
-        $response = $request->executeWithClient($this->getClient());
-        $result = $request->mapResponse($response);
+                $oldName = $project->getName();
+                $name = "new-name-" . ProjectFixture::uniqueProjectString();
 
-        $this->assertInstanceOf(Project::class, $result);
-        $this->assertSame($name, $result->getName());
+                $request = RequestBuilder::of()->project()->update($project)
+                    ->addAction(ProjectChangeNameAction::ofName($name));
+                $response = $this->execute($client, $request);
+                $result = $request->mapFromResponse($response);
 
-        $request = ProjectUpdateRequest::ofVersion($result->getVersion());
-        $request->addAction(ProjectChangeNameAction::ofName($oldName));
-        $response = $request->executeWithClient($this->getClient());
+                $this->assertInstanceOf(Project::class, $result);
+                $this->assertSame($name, $result->getName());
 
-        $this->assertFalse($response->isError());
+                $request = RequestBuilder::of()->project()->update($result)
+                    ->addAction(ProjectChangeNameAction::ofName($oldName));
+                $response = $this->execute($client, $request);
+                $result = $request->mapFromResponse($response);
+
+                $this->assertInstanceOf(Project::class, $result);
+                $this->assertSame($oldName, $result->getName());
+            }
+        );
+
```

Comment:
I left the old code until @jenschude confirmations about the ProjectFixture logic